c7n/filters/core.py

```diff
@@ -352,6 +352,20 @@ def process_value_type(self, sentinel, value):
             # greater than the sentinel typically. Else the syntax for age
             # comparisons is intuitively wrong.
             return value, sentinel
+        elif self.vtype == 'age_in_days':
+            now = datetime.now(tz=tzutc())
+            if not isinstance(sentinel, int):
+                try:
+                    sentinel = int(sentinel)
+                except ValueError:
+                    sentinel = 0
+            if not isinstance(value, datetime):
```

Comment:
this feels like a block that could be extracted into a private utility function rather than repeating multiple times.